### ISSUE TITLE
Refactor C++ into library called by python

### ISSUE THREAD

**Issue by sbailey** [desihub/fiberassign#136]
Getting this work by @tskisner into a ticket for tracking status and questions:

Remove all I/O from the C++ code and make it into a pure library that can be called from python, instead of having a python wrapper that calls the C++ executable to generate temporary files that get read back in to write the final ones.  In the short term this is a fairly major change, but in the long term it should greatly simplify I/O maintainability to have it all in the python while keeping the algorithms in C++.  Ted is exploring this approach in a branch to see if there are any gotchas or whether it is a viable approach.

Related tickets:
* #96 (propagating input target information)
* #108 (optional input config file)
* #113 (robustness to user error on input options)
* #120 (extracting standards from MTL instead of requiring separate file)

**Comment by tskisner** [comment]
Thanks for opening this @sbailey .  For now, I have created a branch to test building all the C++ code directly as a python extension (the "pybind" branch).  This means no more Makefile, just a desi-compatible setup.py.  And instead of calling an external executable, the fiberassign script calls a function from the package.

I verified that a small test data set worked on this branch.  This was the first step in the refactor.  I'm now working on simplifying the internal classes to make it easier to pass data to / from python, which is a requirement for moving the I/O into python.
